### ISSUE TITLE
Error: fork/exec /usr/bin/conmon: permission denied

### ISSUE THREAD

**Issue by lsm5** [containers/container-selinux#170]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Netavark integration tests show a lot of `Error: fork/exec /usr/bin/conmon: permission denied`

```
$ sudo ./bin/podman pod create --name testpod

$ sudo ./bin/podman create IntTest1

$ sudo ./bin/podman create --network IntTest1 --name testpod2

$ sudo ./bin/podman run --name con1 --network IntTest1 alpine nslookup con1
```

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.0.0-dev
API Version:  4.0.0-dev
Go Version:   go1.18beta2
Git Commit:   609c41f86d987fa11e040d0b0a757a5e2844b6c1-dirty
Built:        Tue Feb  8 13:53:02 2022
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.24.1
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-1.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpus: 12
  distribution:
    distribution: fedora
    version: "36"
  eventLogger: journald
  hostname: naruto.nanadai.me
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.17.0-0.rc3.89.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 8834793472
  memTotal: 33399545856
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.2-2.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.2
      commit: f6fbc8f840df1a414f31a60953ae514fa497c748
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 20h 6m 52.78s (Approximately 0.83 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/lsm5/.config/containers/storage.conf
  containerStore:
    number: 47
    paused: 0
    running: 0
    stopped: 47
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/lsm5/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 22
  runRoot: /run/user/1000/containers
  volumePath: /home/lsm5/.local/share/containers/storage/volumes
version:
  APIVersion: 4.0.0-dev
  Built: 1644346382
  BuiltTime: Tue Feb  8 13:53:02 2022
  GitCommit: 609c41f86d987fa11e040d0b0a757a5e2844b6c1-dirty
  GoVersion: go1.18beta2
  OsArch: linux/amd64
  Version: 4.0.0-dev
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Fedora rawhide

**Comment by lsm5** [comment]
@rhatdan works in permissive mode.

```
Feb 09 10:16:20 nagato.nanadai.me audit[113175]: AVC avc:  denied  { entrypoint } for  pid=113175 comm="podman" path="/usr/bin/conmon" dev="dm-1" ino=29262119 scontext=unconfined_u:unconfined_r:unconfined_t:s0 tcontext=system_u:object_r:conmon_exec_t:s0 tclass=file permissive=1
```

**Comment by rhatdan** [comment]
Fixed in https://github.com/containers/container-selinux/releases/tag/v2.177.0

**Comment by rhatdan** [comment]
The quick fix before you can get this package, is to add

chcon -t container_runtime_exec_t bin/podman

Then it should work fine.

**Comment by garrett** [comment]
I've upgraded my system and it pulled in `container-selinux-2.177.0` which actually seems to _cause_ this permission error. No containers run in SELinux enforcing mode now, but they worked in version of container-selinux that was on Fedora 35 last week (`2.173.1-1.fc35`).

Specifically, I  upgraded Silverblue this morning from 35.20220217.0 (2022-02-17T02:42:26Z) to 35.20220222.0 (2022-02-22T03:30:21Z) and changed these packages:

```
$ rpm-ostree db diff 032a54c9c6d3d0aa348ae5a43e2f69465a63a96db84e6aaa4984ab55c06c15e7 83726b81a2e5d5a2464dbdcf2b34d74bf6ca022dab281ca361adbcdfa6d2efec
ostree diff commit from: 032a54c9c6d3d0aa348ae5a43e2f69465a63a96db84e6aaa4984ab55c06c15e7
ostree diff commit to:   83726b81a2e5d5a2464dbdcf2b34d74bf6ca022dab281ca361adbcdfa6d2efec
Upgraded:
  audit 3.0.7-1.fc35 -> 3.0.7-2.fc35
  audit-libs 3.0.7-1.fc35 -> 3.0.7-2.fc35
  btrfs-progs 5.16.1-1.fc35 -> 5.16.2-1.fc35
  container-selinux 2:2.173.1-1.fc35 -> 2:2.177.0-1.fc35
  containers-common 4:1-41.fc35 -> 4:1-45.fc35
  flatpak 1.12.4-1.fc35 -> 1.12.5-1.fc35
  flatpak-libs 1.12.4-1.fc35 -> 1.12.5-1.fc35
  flatpak-selinux 1.12.4-1.fc35 -> 1.12.5-1.fc35
  flatpak-session-helper 1.12.4-1.fc35 -> 1.12.5-1.fc35
  gnome-initial-setup 41.2-1.fc35 -> 41.4-1.fc35
  gnome-user-docs 41.1-1.fc35 -> 41.2-1.fc35
  gtk4 4.4.1-1.fc35 -> 4.4.2-1.fc35
  initscripts-service 10.14-1.fc35 -> 10.15-1.fc35
  libgee 0.20.4-2.fc35 -> 0.20.5-1.fc35
  libibverbs 38.1-2.fc35 -> 39.0-1.fc35
  librsvg2 2.52.5-1.fc35 -> 2.52.6-1.fc35
  osinfo-db 20211216-1.fc35 -> 20220214-1.fc35
  pcsc-lite-ccid 1.4.36-2.fc35 -> 1.5.0-1.fc35
  pipewire 0.3.45-2.fc35 -> 0.3.47-1.fc35
  pipewire-alsa 0.3.45-2.fc35 -> 0.3.47-1.fc35
  pipewire-gstreamer 0.3.45-2.fc35 -> 0.3.47-1.fc35
  pipewire-jack-audio-connection-kit 0.3.45-2.fc35 -> 0.3.47-1.fc35
  pipewire-libs 0.3.45-2.fc35 -> 0.3.47-1.fc35
  pipewire-pulseaudio 0.3.45-2.fc35 -> 0.3.47-1.fc35
  pipewire-utils 0.3.45-2.fc35 -> 0.3.47-1.fc35
  polkit 0.120-1.fc35.1 -> 0.120-1.fc35.2
  polkit-libs 0.120-1.fc35.1 -> 0.120-1.fc35.2
  python3-audit 3.0.7-1.fc35 -> 3.0.7-2.fc35
  qt5-qtwayland 5.15.2-18.fc35 -> 5.15.2-21.fc35
  spice-vdagent 0.21.0-5.fc35 -> 0.22.1-1.fc35
  thermald 2.4.8-1.fc35 -> 2.4.8-3.fc35
  uresourced 0.4.0-2.fc35 -> 0.4.1-1.fc35
  webkit2gtk3 2.34.5-1.fc35 -> 2.34.6-1.fc35
  webkit2gtk3-jsc 2.34.5-1.fc35 -> 2.34.6-1.fc35
```

Then I tried a simple test with `podman`:

```sh
$ podman run -it fedora sh
Error: fork/exec /usr/bin/conmon: permission denied
```

This is with the latest available released version of `container-selinux` on Fedora 35.

```sh
$ rpm -q container-selinux
container-selinux-2.177.0-1.fc35.noarch
```

**Comment by ecocode** [comment]
I have the exact same issue

**Comment by rhatdan** [comment]
Update you container-selinux package

**Comment by ecocode** [comment]
yeah, I am on fedora silverblue and the upgrade I tested was `container-selinux 2:2.173.1-1.fc35 -> 2:2.177.0-1.fc35`.  version 173 is ok, version 177 (being the latest available on this distro) fails.

I will probably have to wait for next version to be available

**Comment by lsm5** [comment]
@ecocode can you fetch packages from updates-testing on silverblue? There should be 2.179.1 but I've now sent v2.180.0.

@rhatdan let's wait a week before the next container-selinux release so that v2.180.0 can land into updates.

**Comment by ecocode** [comment]
Hi,

As I don't know how to get only one package from updates-testing with silverblue, I waited for version 180 to appear in the updates repository.  As it did, I tried it out, but got the same problem.

**Comment by garrett** [comment]
I managed to resolve this Podman issue locally on my Silverblue installation by resetting SELinux on my system:

https://docs.fedoraproject.org/en-US/fedora-silverblue/troubleshooting/#_selinux_problems

I didn't specifically change any SELinux settings, but I do test the SELinux page in Cockpit (for development reasons) from time to time, so it's likely I did something which prevented the newer SELinux defaults from taking effect. Everyone else hitting this issue _probably_ did something to modify SELinux configuration too, whether intentional or not.

If you're having this problem on Silverblue still, you _might_ want to follow the SELinux reset above and reboot your computer for it to take effect.
